### PR TITLE
#BE-4293 Added Base VM Cautions

### PR DESCRIPTION
**Pull request by burakberkkeskin**

### CODE REVIEW

File: docs/self-hosted-appcircle/self-hosted-runner/runner-vm-setup.md [hunk 1]
```diff
@@ -510,10 +510,18 @@ tart clone macOS_240514 vm01
   </TabItem>
 </Tabs>
 
+:::tip
+It's not recommended to delete the base image (`macOS_240514`) as it won't save disk space due to copy-on-write file system on macOS. You can safely re-create `vm01` from the same base image `macOS_240514` without downloading and extracting again from network if needed.
```
Comment (hiilhan) [inline docs/self-hosted-appcircle/self-hosted-runner/runner-vm-setup.md:514]: We may generalize the base image name like `macOS_YYMMDD`, `macOS_<version-code>`. Or indicating it is an example: 
> It's not recommended to delete the base image (e.g., `macOS_240514`) as it won't save disk space due to copy-on-write file system on macOS. You can safely re-create `vm01` from the same base image (e.g., `macOS_240514`) without downloading and extracting again from network if needed.

**Review by hiilhan — CHANGES_REQUESTED** [review]

File: docs/self-hosted-appcircle/self-hosted-runner/runner-vm-setup.md [hunk 2]
```diff
@@ -1267,3 +1275,26 @@ Below step in [update base images](#update-base-images) section,
...
+To list the installed Xcode simulator runtimes, run the following command on your base VM:
+
+```bash
+xcrun simctl runtime list 2>/dev/null
+```
+
+If you determine that certain watchOS or tvOS runtimes are not needed, you can delete them to free up disk space:
+
+```bash
+xcrun simctl delete <runtime_id>
```
Comment (csonuryilmaz) [inline docs/self-hosted-appcircle/self-hosted-runner/runner-vm-setup.md:1293]:
```
xcrun simctl runtime delete <runtime_id>
```

File: docs/self-hosted-appcircle/self-hosted-runner/runner-vm-setup.md [hunk 3]
```diff
@@ -1267,3 +1275,26 @@ Below step in [update base images](#update-base-images) section,
...
+
+Occasionally, you may need to manage disk space on your macOS base VM due to storage constraints or other reasons. One way to free up disk space is by deleting unused Xcode simulator runtimes.
+
+To list the installed Xcode simulator runtimes, run the following command on your base VM:
+
+```bash
+xcrun simctl runtime list 2>/dev/null
+```
+
+If you determine that certain watchOS or tvOS runtimes are not needed, you can delete them to free up disk space:
```
Comment (csonuryilmaz) [inline docs/self-hosted-appcircle/self-hosted-runner/runner-vm-setup.md:1290]: iOS, watchOS, tvOS or visionOS

File: docs/self-hosted-appcircle/self-hosted-runner/runner-vm-setup.md [hunk 4]
```diff
@@ -510,10 +510,18 @@ tart clone macOS_240514 vm01
...
+It's not recommended to delete the base image (`macOS_240514`) as it won't save disk space due to copy-on-write file system on macOS. You can safely re-create `vm01` from the same base image `macOS_240514` without downloading and extracting again from network if needed. 
+:::
+
 In docker terminology, `vm01` and `vm02` will be our docker images. We will configure them separately, persist our changes and then create containers to execute build pipelines. On every build, fresh containers will be used for both runners.
 
 ### Configure Base Runner VMs
 
+Be cautious when updating the base VMs (`vm01` and `vm02`). Any changes made on these base VMs are persisted and may affect disk usage, keychain, and cache files on the runner VMs created from them.
+
+If you're freshly creating the base VMs, you can ignore this warning. However, if you've already registered runners to your Appcircle server and want to make updates to the base VMs, it's highly recommended to [disable the runner](/docs/self-hosted-appcircle/self-hosted-runner/configure-runner/manage-runners.md#monitoring-self-hosted-runners) from the Appcircle dashboard to prevent builds from running on the base VMs.
```
Comment (csonuryilmaz) [inline docs/self-hosted-appcircle/self-hosted-runner/runner-vm-setup.md:523]: `docs` ❌

**Pull request merged**